frontend/src/metabase/setup/components/SettingsPage/SettingsPage.tsx

```diff
@@ -1,41 +1,44 @@
 import LogoIcon from "metabase/components/LogoIcon";
-import { isNotFalsy } from "metabase/lib/types";
 import { useSelector } from "metabase/lib/redux";
-import { getUsageReason } from "metabase/setup/selectors";
+import { getSteps } from "metabase/setup/selectors";
 import { CloudMigrationHelp } from "../CloudMigrationHelp";
-import { DatabaseStep } from "../DatabaseStep";
 import { CompletedStep } from "../CompletedStep";
+import { DatabaseHelp } from "../DatabaseHelp";
+import { DatabaseStep } from "../DatabaseStep";
 import { LanguageStep } from "../LanguageStep";
 import { PreferencesStep } from "../PreferencesStep";
-import { UserStep } from "../UserStep";
 import { SetupHelp } from "../SetupHelp";
-import { DatabaseHelp } from "../DatabaseHelp";
+import type { NumberedStepProps } from "../types";
 import { UsageQuestionStep } from "../UsageQuestionStep";
+import { UserStep } from "../UserStep";
 import { PageBody, PageHeader } from "./SettingsPage.styled";
 
-export const SettingsPage = (): JSX.Element => {
-  const usageReason = useSelector(getUsageReason);
+const STEP_COMPONENTS: Record<
+  string,
+  (props: NumberedStepProps) => React.ReactElement
+> = {
+  language: LanguageStep,
+  user_info: UserStep,
+  usage_question: UsageQuestionStep,
+  db_connection: DatabaseStep,
+  data_usage: PreferencesStep,
```

Comment:
I totally agree, since it's a bit of files to change though, I'd rather do it in a PR towards master after this all of the rest of the epic gets merged.
I added a subtask to the epic so I don't forget about it

Reply:
Sounds good.